### PR TITLE
PeerGroup: migrate `vConnectTimeout` field to `java.time` API

### PR DESCRIPTION
**Pull request by schildbach**

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/core/PeerGroup.java [hunk 1]
```diff
@@ -1128,7 +1128,7 @@ private boolean maybeCheckForLocalhostPeer() {
             // Do a fast blocking connect to see if anything is listening.
             try (Socket socket = new Socket()) {
                 socket.connect(new InetSocketAddress(InetAddress.getLoopbackAddress(), params.getPort()),
-                        vConnectTimeoutMillis);
+                        (int) vConnectTimeout.toMillis());
```
Comment (msgilligan) [inline core/src/main/java/org/bitcoinj/core/PeerGroup.java:1131]: Should you use `Math.toIntExact()` here to be safe?

File: core/src/main/java/org/bitcoinj/net/BlockingClient.java [hunk 2]
```diff
@@ -78,7 +78,7 @@ public BlockingClient(final SocketAddress serverAddress, final StreamConnection
             if (clientSet != null)
                 clientSet.add(BlockingClient.this);
             try {
-                socket.connect(serverAddress, connectTimeoutMillis);
+                socket.connect(serverAddress, (int) connectTimeout.toMillis());
```
Comment (msgilligan) [inline core/src/main/java/org/bitcoinj/net/BlockingClient.java:81]: `Math.toIntExact()`?

**Review by msgilligan — COMMENTED** [review]
Looks good. I suggested in two places that you _consider_ using `Math.toIntExact()` just to be safe. I haven't thought about the likelihood (or impossibility) of that ever triggering a RTE, though.

File: core/src/main/java/org/bitcoinj/core/PeerGroup.java (hunk 1 above)
Comment (schildbach) [reply core/src/main/java/org/bitcoinj/core/PeerGroup.java]: Good idea! Pushed a commit on top.

**Review by msgilligan — APPROVED** [review]
Looks good!

**Comment by schildbach** [comment]
Merged.

**Pull request closed without merging**